src/globus_cli/commands/gcp/set_subscription_id.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+
+from globus_cli.constants import EXPLICIT_NULL, ExplicitNullType
+from globus_cli.endpointish import Endpointish, EntityType
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.termio import TextMode, display
+
+
+class GCPSubscriptionIdType(click.ParamType):
+    def get_type_annotation(self, _: click.Parameter) -> type:
+        return t.cast(type, uuid.UUID | ExplicitNullType)
+
+    def convert(
+        self, value: str, param: click.Parameter | None, ctx: click.Context | None
+    ) -> t.Any:
```

Comment:
It's not super-closely related to this PR, but just because I think it's cool, I wanted to mention that under the latest `click-type-test`, you can write `convert(...) -> uuid.UUID | ExplicitNullType` and omit `get_type_annotation`.

Reply:
With #923 merged, you could now apply this change here (after a rebase).